test/distributed/fsdp/test_fsdp_optim_state.py

```diff
@@ -446,8 +446,7 @@ def test_full_optim_state_dict_nested(
             full_osd, ref_osd, check_same_param_keys=check_same_param_keys,
         )
 
-    # Require 4 GPUs since we test halving the world size
-    @skip_if_lt_x_gpu(4)
+    @skip_if_lt_x_gpu(2)
```

Comment:
To make sure these tests always run, we decrease the GPU requirement to 2. This means that the tests that halve the world size may halve from 2 to 1, in which case the parameters are no longer sharded. This is still a meaningful exercise of the saving and loading.

The halving of the world size is done by only including even ranks, meaning that the test's implementation is adaptable to different total number of GPUs (e.g. whether 2 or 4).